sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -152,6 +154,16 @@ abstract class Optimizer(sessionCatalog: SessionCatalog, conf: SQLConf)
 }
 
 /**
+ * Remove useless DISTINCT for MAX and MIN
```

Comment:
Also need to emphasize `EliminateDistinct ` should be before `ReplaceDeduplicateWithAggregate`